### PR TITLE
Watersmart 525

### PR DESCRIPTION
**Pull request by mbucknell**
Made a new view StreamflowPlotView which manages the display of the plot loading indicator and creates the plot and plot legend. This is used in the StreamflowHucDataStatsView replacing existing code and added to StreamflowGageDataStatsView.

The large number of changed files is due to removing dygraphs and instead using webjars to get this dependancy. I have also upgraded dygraphs to version 1.1.1

### CODE REVIEW

File: nwc/src/main/webapp/js/utils/jqueryUtils.js [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+ */
+NWC.util.findXMLNamespaceTags = function($xml, nsTag) {
+	"use strict";
+	var tag = nsTag.substr(nsTag.indexOf(':') + 1);
+	var nsEscTag = nsTag.replace(':', '\\:');
+
+	return $xml.find(nsEscTag + ', ' + tag);
+};
+
+
```
Comment (mbucknell) [inline nwc/src/main/webapp/js/utils/jqueryUtils.js]: Copied from the GDP project.

File: nwc/src/main/webapp/js/view/StreamflowPlotView.js [hunk 2]
```diff
@@ -0,0 +1,62 @@
+/*jslint browser: true*/
+
+var NWC = NWC || {};
+
+NWC.view = NWC.view || {};
+
+(function() {
+	"use strict";
```
Comment (thongsav-usgs) [inline nwc/src/main/webapp/js/view/StreamflowPlotView.js]: Heh, I was just wondering how I was going to wrap my extended classes to include "use strict". I noted this wasn't used in the class below this one.

File: nwc/src/main/webapp/js/view/StreamflowStatsGageDataView.js [hunk 3]
```diff
@@ -181,6 +211,83 @@ NWC.view.StreamflowStatsGageDataView = NWC.view.BaseStreamflowStatsDataView.exte
...
+			alert('Retrieving data for this plot failed with error: ' + args[0]);
+		});
+	},
+
+	remove : function() {
+		this.streamflowPlotView.remove();
+		NWC.view.BaseStreamflowStatsDataView.prototype.remove.apply(this, arguments);
+	},
+
+	_buildStreamFlowUrl : function(startDate, endDate, siteId) {
```
Comment (thongsav-usgs) [inline nwc/src/main/webapp/js/view/StreamflowStatsGageDataView.js]: Any reason you are building the url manually and not just letting $,ajax build it via serializing the "data" attribute?

File: nwc/src/main/webapp/js/view/StreamflowStatsHucDataView.js [hunk 4]
```diff
@@ -14,15 +17,30 @@ NWC.view.StreamflowStatsHucDataView = NWC.view.BaseStreamflowStatsDataView.exten
...
+		'click .download-streamflow-btn' : 'downloadData'
 	},
 
 	render : function() {
 		NWC.view.BaseStreamflowStatsDataView.prototype.render.apply(this, arguments);
+
 		this.map.render(this.insetMapDiv);
+		this.streamflowPlotView = new NWC.view.StreamflowPlotView({
+			el : this.$el.find('.streamflow-plot-container'),
+			getDataSeriesPromise : this.getDataSeriesPromise.bind(this)
```
Comment (thongsav-usgs) [inline nwc/src/main/webapp/js/view/StreamflowStatsHucDataView.js]: Didn't know about this bind function, will come in handy.

File: nwc/src/main/webapp/js/view/WaterBudgetHucDataView.js [hunk 5]
```diff
@@ -159,7 +158,7 @@ NWC.view.WaterBudgetHucDataView = NWC.view.BaseView.extend({
         var labels = this.dataSeriesStore[plotTimeDensity].getSeriesLabelsAs(
                 measurementSystem, normalization, plotTimeDensity);
         var ylabel = NWC.util.Units[measurementSystem][normalization][plotTimeDensity];
-        NWC.util.Plotter.getPlot(plotDivSelector, legendDivSelector, values, labels, ylabel);
+        NWC.util.Plotter.getPlot($('#waterBudgetPlot'), $('#waterBudgetLegend'), values, labels, ylabel);
```
Comment (thongsav-usgs) [inline nwc/src/main/webapp/js/view/WaterBudgetHucDataView.js]: *cough* spaces *gag* :smiling_imp:

**Comment by thongsav-usgs** [comment]
:+1:

File: nwc/src/main/webapp/js/view/StreamflowPlotView.js (hunk 2 above)
Comment (mbucknell) [inline nwc/src/main/webapp/js/view/StreamflowPlotView.js]: Credit goes to @cschroed-usgs for coming up with this pattern. I'm trying to change existing ones as I go.

File: nwc/src/main/webapp/js/view/StreamflowStatsGageDataView.js (hunk 3 above)
Comment (mbucknell) [inline nwc/src/main/webapp/js/view/StreamflowStatsGageDataView.js]: When I first created this function, I did not have the url in the config. Now that it is, I can change the function to return the data attribute.

File: nwc/src/main/webapp/js/view/WaterBudgetHucDataView.js (hunk 5 above)
Comment (mbucknell) [inline nwc/src/main/webapp/js/view/WaterBudgetHucDataView.js]: This is existing code. There are spaces in many parts of the app, especially the utility packages.

File: nwc/src/main/webapp/js/view/StreamflowStatsHucDataView.js (hunk 4 above)
Comment (mbucknell) [inline nwc/src/main/webapp/js/view/StreamflowStatsHucDataView.js]: Underscore also has a _.bind function.

**Pull request merged**